executor/internal/calibrateresource/calibrate_resource.go

```diff
@@ -344,9 +372,46 @@ func (e *Executor) dynamicCalibrate(ctx context.Context, req *chunk.Chunk, exec
 	for i := lowerBound; i < upperBound; i++ {
 		sum += quotas[i]
 	}
-	quota := sum / float64(upperBound-lowerBound)
-	req.AppendUint64(0, uint64(quota))
-	return nil
+	return sum / float64(upperBound-lowerBound), nil
```

Comment:
Could `upperBound-lowerBound` be zero?

Reply:
It cannot be zero.
because `len(quotas) >= 2`. So `len(quotas)` must be greater than `round(0.1*len(quotas))`